### PR TITLE
Add observedgeneration to ambient statuses

### PR DESCRIPTION
**Pull request by tjons**
**Please provide a description of this PR:**
Display the `ObservedGeneration` of the object in Ambient object conditions. cc @ilrudie

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
I don't get how this works end to end? The `IstioCondition` type does not have `ObservedGeneration` at all

File: go.sum [hunk 1]
```diff
@@ -415,6 +415,8 @@ github.com/stretchr/testify v1.9.0 h1:HtqpIVDClZ4nwg75+f6Lvsy/wHu+3BoSGCbBAcpTsT
 github.com/stretchr/testify v1.9.0/go.mod h1:r2ic/lqez/lEtzL7wO/rwa5dbSLXVDPFyf8C91i36aY=
 github.com/subosito/gotenv v1.6.0 h1:9NlTDc1FTs4qu0DDq7AEtTPNw6SVm7uBMsUCUjABIf8=
 github.com/subosito/gotenv v1.6.0/go.mod h1:Dk4QP5c2W3ibzajGcXpNraDfq2IrhjMIvMSWPKKo0FU=
+github.com/tjons/api v0.0.0-20241022224639-7a6e49ca3102 h1:p0rKANEJjG+NJl8303bZ28X3SbwFLCUyuAfNNOLFt0s=
```
Comment (zirain) [inline go.sum:418]: ???

Comment (tjons) [reply go.sum:418]: this is WIP - untangling merge changes, please bear with me and I will fix shortly!

**Comment by tjons** [comment]
Sorry to all who I accidentally requested reviews from! :) @howardjohn this is tested and works end-to-end. 

To test, apply an authorization policy to an ambient-enrolled namespace:

```
kubectl apply -f - <<EOF
apiVersion: security.istio.io/v1
kind: AuthorizationPolicy
metadata:
  name: productpage-viewer
  namespace: default
spec:
  targetRefs:
  - kind: Service
    group: ""
    name: productpage
  action: ALLOW
  rules:
  - from:
    - source:
        principals:
        - cluster.local/ns/default/sa/sleep
    to:
    - operation:
        methods: ["GET"]
EOF
```

With a waypoint in the namespace, the status should look like this:
```
apiVersion: security.istio.io/v1
kind: AuthorizationPolicy
metadata:
  creationTimestamp: "2024-10-29T03:01:37Z"
  generation: 1
  name: productpage-viewer
  namespace: default
  resourceVersion: "3503"
  uid: f7d2f3c2-105a-4b5e-903e-3a2090f3fc74
spec:
  action: ALLOW
  rules:
  - from:
    - source:
        principals:
        - cluster.local/ns/default/sa/sleep
    to:
    - operation:
        methods:
        - GET
  targetRefs:
  - group: ""
    kind: Service
    name: productpage
status:
  conditions:
  - lastTransitionTime: "2024-10-29T03:03:11.077536482Z"
    message: bound to default/waypoint
    observedGeneration: "1"
    reason: Accepted
    status: "True"
    type: WaypointAccepted
```

Editing the policy will increment the `observedGeneration` in the written condition.

**Review by howardjohn — APPROVED** [review]
:+1:

**Comment by ilrudie** [comment]
do we want to consider this for cherry-pick to 1.24?

**Pull request merged**